### PR TITLE
fix some source and add source

### LINKED ISSUE #215: Lxmanga source request
### Source name

Lxmanga

### Source link

https://lxmanga.net/

### Language

Vietnamese

### Other details

A hentai source in vietnamese

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### PR DESCRIPTION
**Pull request by davvarrr**
fix some source and add 4 source
immortalupdates.id
zahard.xyz
mangakyo.org
lugnica-scans.com
lxmanga.net -> close #215

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		sortOrder: SortOrder,
+	): List<Manga> {
+
+		val url = buildString {
+			append("https://")
+			append(domain)
+			if (sortOrder == SortOrder.ALPHABETICAL) {
+				append("/mangas/")
+				// just to stop the search of the ALPHABETICAL page because it contains all the manga and has no page function ( to change if there is a better method to stop the search )
+				if (page.toString() == "2") {
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt:61]:
```suggestion
				if (page == 2) {
```
??

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+					source = source,
+				)
+			},
+			author = root.select("div.manga-staff").text(),
+			description = root.selectFirst("div.manga-description div")?.text(),
+			chapters = root.select("div.manga-chapters_wrapper div.manga-chapter")
+				.mapChapters(reversed = true) { i, div ->
+
+					val a = div.selectFirstOrThrow("a")
+					val href = a.attr("href")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt:145]:
```suggestion
					val href = a.attrAsRelativeUrl("href")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+						source = source,
+					)
+				},
+		)
+	}
+
+	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+		val fullUrl = chapter.url.toAbsoluteUrl(domain)
+		val doc = webClient.httpGet(fullUrl).parseHtml()
+		val root = doc.body().selectFirst("#forgen_reader")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LugnicaScans.kt:169]:
```suggestion
		val root = doc.body().requireElementById("forgen_reader")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/LxManga.kt [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+		}
+
+
+		val doc = webClient.httpGet(url).parseHtml()
+
+
+		return doc.select("div.grid div.manga-vertical")
+			.map { div ->
+				val href = div.selectFirstOrThrow("a").attr("href")
+				val img = div.select(".cover").first()!!.attr("style").substringAfter("url('").substringBefore("')")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/LxManga.kt:78]:
```suggestion
				val img = div.selectFirstOrThrow(".cover").attr("style").substringAfter("url('").substringBefore("')")
```

**Comment by davvarrr** [comment]
Thanks for the corrections :)

**Pull request merged**